libs/spark/src/Unstable_Button/Unstable_Button.tsx

```diff
@@ -0,0 +1,277 @@
+        '&:hover': {
+          backgroundColor: theme.unstable_palette.red[600],
+        },
+        '&:active': {
+          backgroundColor: darken(theme.unstable_palette.red[700], 0.2),
+        },
+        '&[aria-expanded="true"]': {
+          backgroundColor: theme.unstable_palette.neutral[600],
+        },
+        '&$disabled': {
+          backgroundColor: theme.unstable_palette.neutral[80],
+          color: theme.unstable_palette.neutral[100],
+        },
+      }),
+      ...(props.size === 'small' && {
+        ...buttonFontVariantSmall,
+        padding: '8px 16px',
+      }),
+      ...(props.size === 'medium' && {
+        ...buttonFontVariantMedium,
+        padding: '12px 24px',
+      }),
+      ...(props.size === 'large' && {
+        ...buttonFontVariantLarge,
+        padding: '20px 32px',
+      }),
+    }),
+
+    label: (props: Unstable_ButtonProps) => ({
+      color: 'inherit',
+      fontSize: 'inherit',
+      lineHeight: 'inherit',
+      ...(props.variant === 'primary' && {
+        color: theme.palette.common.white,
+        '&$disabled': {
+          color: theme.unstable_palette.neutral[100],
+        },
+      }),
+      ...(props.variant === 'stroked' && {
+        color: theme.unstable_palette.brand.blue,
+        '&[aria-expanded="true"]': {
+          color: theme.palette.common.white,
+        },
+      }),
+      ...(props.variant === 'ghost' && {
+        color: theme.unstable_palette.brand.blue,
+        '&[aria-expanded="true"]': {
+          color: theme.palette.common.white,
+        },
+        '&$disabled': {
+          color: theme.unstable_palette.neutral[100],
+        },
+      }),
+      ...(props.variant === 'destructive' && {
+        color: theme.palette.common.white,
+        '&$disabled': {
+          color: theme.unstable_palette.neutral[100],
+        },
+      }),
+    }),
+    startIcon: (props: Unstable_ButtonProps) => ({
+      display: 'inherit',
+      // -2px accounts for label vs icon size difference
+      // have to specify individually to override defaults
+      margin: '-2px',
+      marginBottom: '-2px',
+      marginRight: '8px',
+      marginLeft: 0,
+      color: 'inherit',
+      ...(props.size === 'small' && {
+        fontSize: theme.typography.pxToRem(16),
+        lineHeight: theme.typography.pxToRem(16),
+      }),
+      ...(props.size === 'medium' && {
+        fontSize: theme.typography.pxToRem(24),
+        lineHeight: theme.typography.pxToRem(24),
+      }),
+      ...(props.size === 'large' && {
+        fontSize: theme.typography.pxToRem(24),
+        lineHeight: theme.typography.pxToRem(24),
+      }),
+    }),
+    endIcon: (props: Unstable_ButtonProps) => ({
+      display: 'inherit',
+      // -2px accounts for label vs icon size difference
+      // have to specify individually to override defaults
+      marginTop: '-2px',
+      marginBottom: '-2px',
+      marginLeft: '8px',
+      marginRight: 0,
```

Comment:
Okay on revisit, I think I know why you specified this. Is it because the text baseline in Figma is different than the Web so the text seems like it's "too high" in comparison?

If so, here's the workaround I found:

```
{
  label: { marginTop: 2, marginBottom: -2 }, // shift text baseline down
  startIcon: { marginTop: -2 }, // shift icon baseline up
  endIcon: { marginTop: -2 }
}
```

Reply:
Some references: 
- Figma with an arbitrary box to show the extension of the "baseline" and the line of "cap-height"
- Implementation before margin adjustments
- and after
![image](https://user-images.githubusercontent.com/25781782/162764370-a2b2dcc7-8862-4591-a8db-b73967e12bfb.png)
![image](https://user-images.githubusercontent.com/25781782/162764384-e57b6b79-98e1-49a6-9c6d-169cf9a367b7.png)
![image](https://user-images.githubusercontent.com/25781782/162764391-4fbfc4e7-62b2-4eb0-b9d4-c63e714df0fa.png)